src/core/sys/solaris/sys/elf.d

```diff
@@ -363,165 +54,28 @@ enum SHT_SUNW_SIGNATURE = 0x6ffffff6;
-
-enum SHN_UNDEF       = 0;
-enum SHN_LORESERVE   = 0xff00;
-enum SHN_LOPROC      = 0xff00;
-enum SHN_HIPROC      = 0xff1f;
-enum SHN_LOOS        = 0xff20;
-enum SHN_LOSUNW      = 0xff3f;
-enum SHN_SUNW_IGNORE = 0xff3f;
-enum SHN_HISUNW      = 0xff3f;
-enum SHN_HIOS        = 0xff3f;
-enum SHN_ABS         = 0xfff1;
-enum SHN_COMMON      = 0xfff2;
-enum SHN_XINDEX      = 0xffff;
-enum SHN_HIRESERVE   = 0xffff;
-
-struct Elf32_Sym
-{
-    Elf32_Word    st_name;
-    Elf32_Addr    st_value;
-    Elf32_Word    st_size;
-    ubyte         st_info;
-    ubyte         st_other;
-    Elf32_Half    st_shndx;
-}
-
-struct Elf64_Sym
-{
-    Elf64_Word    st_name;
-    ubyte         st_info;
-    ubyte         st_other;
-    Elf64_Half    st_shndx;
-    Elf64_Addr    st_value;
-    Elf64_Xword   st_size;
-}
-
-enum STN_UNDEF  = 0;
-
-extern (D)
-{
-    auto ELF32_ST_BIND(T)(T val) { return cast(ubyte)val >> 4; }
-    auto ELF32_ST_TYPE(T)(T val) { return val & 0xf; }
-    auto ELF32_ST_INFO(B, T)(B bind, T type) { return (bind << 4) + (type & 0xf); }
-    alias ELF32_ST_BIND ELF64_ST_BIND;
-    alias ELF32_ST_TYPE ELF64_ST_TYPE;
-    alias ELF32_ST_INFO ELF64_ST_INFO;
-}
-
-enum STB_LOCAL  = 0;
-enum STB_GLOBAL = 1;
-enum STB_WEAK   = 2;
-enum STB_NUM    = 3;
-enum STB_LOPROC = 13;
-enum STB_HIPROC = 15;
-
-enum STT_NOTYPE  = 0;
-enum STT_OBJECT  = 1;
-enum STT_FUNC    = 2;
-enum STT_SECTION = 3;
-enum STT_FILE    = 4;
-enum STT_COMMON  = 5;
-enum STT_TLS     = 6;
-enum STT_NUM     = 7;
-enum STT_LOOS    = 10;
-enum STT_HIOS    = 12;
-enum STT_LOPROC  = 13;
-enum STT_HIPROC  = 15;
 
 extern (D)
 {
     auto ELF32_ST_VISIBILITY(O)(O o) { return o & 0x07; }
     alias ELF32_ST_VISIBILITY ELF64_ST_VISIBILITY;
 }
 
-enum STV_DEFAULT   = 0;
-enum STV_INTERNAL  = 1;
-enum STV_HIDDEN    = 2;
-enum STV_PROTECTED = 3;
 enum STV_EXPORTED  = 4;
 enum STV_SINGLETON = 5;
 enum STV_ELIMINATE = 6;
 enum STV_NUM       = 7;
 
-struct Elf32_Rel
-{
-    Elf32_Addr    r_offset;
-    Elf32_Word    r_info;
-}
-
-struct Elf32_Rela
-{
-    Elf32_Addr    r_offset;
-    Elf32_Word    r_info;
-    Elf32_Sword   r_addend;
-}
-
-struct Elf64_Rel
-{
-    Elf64_Addr    r_offset;
-    Elf64_Xword   r_info;
-}
-
-struct Elf64_Rela
-{
-    Elf64_Addr    r_offset;
-    Elf64_Xword   r_info;
-    Elf64_Sxword  r_addend;
-}
-
 extern (D)
 {
-    auto ELF32_R_SYM(V)(V val) { return val >> 8; }
-    auto ELF32_R_TYPE(V)(V val) { return val & 0xff; }
-    auto ELF32_R_INFO(S, T)(S sym, T type) { return (sym << 8) + (type & 0xff); }
-
-    auto ELF64_R_SYM(I)(I i) { return i >> 32; }
-    auto ELF64_R_TYPE(I)(I i) { return i & 0xffffffff; }
-    auto ELF64_R_INFO(S, T)(S sym, T type) { return (sym << 32) + (type); }
-
     auto ELF64_R_TYPE_DATA(I)(I i) { return (i << 32) >> 40; }
     auto ELF64_R_TYPE_ID(I)(I i) { return (i << 56) >> 56; }
     auto ELF64_R_TYPE_INFO(S, T)(S sym, T type) { return (sym <<8) + (type); }
 }
 
-enum GRP_COMDAT = 0x01;
-
 struct Elf32_Nhdr
 {
     Elf32_Word n_namesz;
```

Comment:
For reviewers:
Regarding ELF notes, while there might be some common note sections, these are largely platform-specific.